anndata/_io/h5ad.py

```diff
@@ -522,19 +523,20 @@ def read_group(group: h5py.Group) -> Union[dict, pd.DataFrame, sparse.spmatrix]:
     if "h5sparse_format" in group.attrs:  # Backwards compat
         return SparseDataset(group).to_memory()
 
-    encoding_type = group.attrs.get("encoding-type")
-    if encoding_type:
+    try:
```

Comment:
I would rather check that that the key existed than be permissive about any KeyError. Could this be done with `if "encoding-type" in group.attrs` instead?